lib/chatRoom.js

```diff
@@ -0,0 +1,33 @@
+const shortId = require('shortid');
+class ChatRoom {
+  constructor() {
+    this.clients = new Map(); // store usernames here    
+  }
+
+  add(client) {
+    const username = shortId.generate();
+    client.username = `'user ' + ${username}`;
+    this.clients.set(username, client);
+    return client;
+  }
+
+  getClient(username) {
+    const client = this.clients.get(username); //client?? also 19
+    return client;
+  }
+
+  rename(oldName, chosenName) {
+    const client = this.clients.get(oldName);
+    const deleted = this.clients.delete(oldName);
+    this.clients.username = chosenName;
+    this.clients.set(chosenName, client);
+    return deleted;
+  }
+
+  all(clients) {
```

Comment:
```suggestion
  all() {
     return [...this.clients.values()];
  }
```
Per the [lab.md](https://github.com/alchemy-fullstack-js-winter-2019/tcp-server/blob/master/LAB.md), the 'all' method shouldn't require anything to be passed to it and I would expect it to return an array like the above. Take a look at Ryan's example code for a guide on testing this function.